db/collection_update.go

```diff
@@ -364,13 +365,45 @@ func (c *collection) patchPrimaryDoc(
 	}
 	primaryCol = primaryCol.WithTxn(txn)
 
-	primaryField, _ := primaryCol.Description().GetRelation(relationFieldDescription.RelationName)
+	primaryField, ok := primaryCol.Description().GetRelation(relationFieldDescription.RelationName)
+	if !ok {
+		return client.NewErrFieldNotExist(relationFieldDescription.RelationName)
+	}
+
+	primaryIDField, ok := primaryCol.Description().Schema.GetField(primaryField.Name + request.RelatedObjectID)
+	if !ok {
+		return client.NewErrFieldNotExist(primaryField.Name + request.RelatedObjectID)
+	}
 
-	_, err = primaryCol.UpdateWithKey(
+	doc, err := primaryCol.Get(
 		ctx,
 		primaryDockey,
-		fmt.Sprintf(`{"%s": "%s"}`, primaryField.Name+request.RelatedObjectID, docKey),
+		false,
 	)
+	if err != nil && !errors.Is(err, ds.ErrNotFound) {
```

Comment:
question: Shouldn't we return an error if we are trying to set a relation that doesn't exists?

Reply:
We don't do this at all for anything at the moment and I would rather we added such behaviour deliberately and consistently instead of piecemeal as we happen upon such locations in the codebase

Reply:
Alright. Just wanted to make sure that that is what we should be doing. Certainly out of scope here.